sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,120 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
+    the top `k` labels predicted (ranked by predicted scores). Note that the
+    multilabel and multioutput cases aren't covered here.
+
+    Read more in the :ref:`User Guide <top_k_accuracy_score>`
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels. Expected to be a sequence of int in the range
+        (0, n_classes - 1).
+
+    y_score : array-like of shape (n_samples, n_classes)
+        Target scores. These can be either probability estimates or
+        non-thresholded decision values as returned by :term:`predict_proba` or
+        :term:`decision_function`, respectively.
+
+    k : int, default=3
+        Number of guesses allowed to find the correct label.
+
+    normalize : bool, default=True
+        If `True`, return the fraction of correctly classified samples.
+        Otherwise, return the number of correctly classified samples.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights. If `None`, all samples are given the same weight.
+
+    Returns
+    -------
+    score : float
+        The top-k accuracy score. The best performance is 1 with
+        `normalize == True` and the number of samples with
+        `normalize == False`.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    In cases where two or more labels are assigned equal predicted scores,
+    the labels with the smallest indices will be chosen first. Then, the result
+    might be incorrect if the correct label falls after the threshold because
+    of that.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import top_k_accuracy_score
+    >>> y_true = np.array([0, 1, 2, 2])
+    >>> y_score = np.array([[0.5, 0.2, 0.2],
+    ...                     [0.3, 0.4, 0.2],
+    ...                     [0.2, 0.4, 0.3],
+    ...                     [0.7, 0.2, 0.1]])
+    >>> top_k_accuracy_score(y_true, y_score, k=2)
+    0.75
+    >>> top_k_accuracy_score(y_true, y_score, k=2, normalize=False)
+    3
+    >>> top_k_accuracy_score([0, 0, 0, 0], y_score, k=2)
+    0.75
+
+    """
+    if k == 1:
+        raise ValueError(
+            "'k'=1 is equivalent to 'metrics.accuracy_score'. Please, use "
+            "that function instead."
+        )
+
+    check_consistent_length(y_true, y_score, sample_weight)
+    y_true_type = type_of_target(y_true)
+
+    if y_true_type not in {'binary', 'multiclass'}:
+        raise ValueError(
+            "'y_true' type must be 'binary' or 'multiclass', got "
+            f"'{y_true_type}' instead."
+        )
+
+    y_true = column_or_1d(y_true)
+    y_score = check_array(y_score)
+
+    n_classes = y_score.shape[1]
+    if n_classes < 3:
+        raise ValueError(
+            "In a multiclass setting the number of columns in 'y_score' must "
+            f"be greater than 2, got ({n_classes}) instead. Please see "
```

Comment:
Since we're always in a multiclass setting:

```suggestion
            "The number of columns in 'y_score' must "
            f"be strictly greater than 2, got ({n_classes}) instead. Please use "
```